figures/serializers.py

```diff
@@ -749,8 +749,19 @@ class Meta:
 
     def get_registration_fields(self, user):
         registration_fields = dict()
-        for field in self.context['required_fields']:
-            registration_fields[field] = str(getattr(user.profile, field, ''))
+        user_profile = user.profile
+        user_fields = [
+            user_field for user_field in user_profile._meta.fields
+            if user_field.name in self.context['required_fields']
+        ]
+
+        for user_field in user_fields:
```

Comment:
You can ommit above generator and can do it in single loop.
```
for user_field in user_profile._meta.fields:
    if user_field.name in self.context['required_fields']:
       if user_field.choices:
                value = user_profile._get_FIELD_display(user_field)
        else:
               value = getattr(user_profile, user_field.name)
```

Reply:
Pardon my indentation.

Reply:
Done.